Mathlib/Algebra/Module/Torsion.lean

```diff
@@ -866,3 +863,36 @@ theorem isTorsion_iff_isTorsion_int [AddCommGroup M] :
 #align add_monoid.is_torsion_iff_is_torsion_int AddMonoid.isTorsion_iff_isTorsion_int
 
 end AddMonoid
+
+namespace AddSubgroup
+
+variable (A : Type*) [AddCommGroup A] (n : ℤ)
+
+/-- The additive `n`-torsion subgroup for an integer `n`. -/
+@[reducible]
+def torsionBy : AddSubgroup A :=
+  (Submodule.torsionBy ℤ A n).toAddSubgroup
+
+@[inherit_doc]
+scoped notation:max A"["n"]" => torsionBy A n
+
+lemma torsionBy_negative_eq : A[-n] = A[n] := by
+  ext a
+  simp
+
+variable {A} {n : ℕ}
+
+@[simp]
+lemma nsmul_torsionBy (x : A[n]) : n • x = 0 :=
```

Comment:
You've decided to make A[n] a term not a type, so shouldn't this lemma be about `x \in A[n]` rather than promoting it to a type? Or do you think you made a bad decision making it a term, and should instead make it `TorsionBy` and a type?